canvas.go

```diff
@@ -269,28 +269,36 @@ func (c *Canvas) servicesGroup(canvas *svg.SVG) {
 	}
 }
 
-// Marshal renders the SVG to the given io.Writer
-func (c *Canvas) Marshal(w io.Writer) {
-	width, height := c.layout()
+// Compute the scale.
+func (c *Canvas) computeScale(width, height int) float32 {
 	scale := float32(1)
 	if height > maxHeight {
 		scale = maxHeight / float32(height)
 	}
 	if float32(width)*scale > maxWidth {
-		scale = maxHeight / float32(height)
+		scale = maxWidth / float32(width)
```

Comment:
This seemed like an error to me.  Drive-by code fix.

I also created the function computeScale so it could be more easily tested.